h/static/scripts/test/widget-controller-test.js

```diff
@@ -441,6 +443,27 @@ describe('WidgetController', function () {
     });
   });
 
+  describe('deferred websocket connection', function () {
+    it('should connect the websocket the first time the sidebar opens', function () {
+      $rootScope.$emit('sidebarOpened');
+      assert.called(fakeStreamer.connect);
+    });
+
+    describe('when logged in user changes', function () {
+      it('should not reconnect if the sidebar is closed', function () {
```

Comment:
If we remove the `sidebarOpen` check as suggested in https://github.com/hypothesis/client/pull/20#discussion_r71667939 then these tests can be simplified as well.